### DISCUSSION TITLE
PUDL v2024.5.0: 2024Q2 updates & new data for energy modelers

### DISCUSSION THREAD

**Discussion by zaneselvans** [catalyst-cooperative/pudl#3651] [Announcements]
We've just completed our quarterly integration of EIA data sources for 2024Q2 (in support of [RMI's Utility Transition Hub](https://utilitytransitionhub.rmi.org/)) and have also added a bunch of new tables over the last few months in an effort to better support energy system modelers (with support from [GridLab](https://gridlab.org)).

See the PUDL documentation for [instructions on accessing all the data](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_access.html)

Here's an overview of what's new:

# 2024Q2 Quarterly Update

* EIA-860M generator data is available through 2024-03
* EIA-923 monthly updates are available through 2024-02
* EIA bulk electricity data (used for filling in missing prices in the Fuel Receipts and Costs table) is available through 2024-02
* EPA CEMS hourly data is available through 2024-03

Note that the data provided by EIA-860M is only a small subset of what's available in the EIA-860 (just data about generators), and EIA still hasn't published the EIA-860 early release data for 2023, there are many gaps in the output tables for 2023 and 2024, since some of them depend on EIA-860 data that's not available in the EIA-860M.  The early release data for 2023 is expected out in June, 2024.

# New Data for Energy System Modeling

## EIA-860 Wind, Solar, and Storage Attributes

We've added tables detailing generator attributes specific to wind, solar, and energy storage facilities which have been published in the EIA-860 in recent years. For now we aren't merging these additional attributes into the already very wide generic generator attributes table, since it would result in many columns which were almost entirely null. If you need all the attributes for a given wind, solar, or storage facility you'll want to merge them together.

(Links go to the PUDL data dictionary)

* [core_eia860__scd_generators_energy_storage](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-eia860-scd-generators-energy-storage)
* [core_eia860__scd_generators_solar](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-eia860-scd-generators-solar)
* [core_eia860__scd_generators_wind](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-eia860-scd-generators-wind)

## EIA-923 Storage Operations

Similarly, EIA has recently started publishing data describing monthly energy storage facility operations, which can be found in:

* [core_eia923__monthly_energy_storage](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-eia923-monthly-energy-storage)

## EIA-930

We've integrated bulk hourly data from the [EIA Electric Grid Monitor](https://www.eia.gov/electricity/gridmonitor/dashboard/electric_overview/US48/US48) (aka EIA-930).

Note that this data is relatively unprocessed and has mostly just been reshaped and normalized. Depending on your use case, there's lots of additional cleaning and reconciliation of this data that might make sense. See e.g. [gridemissions](https://github.com/jdechalendar/gridemissions/) from @jdechalendar. If you have thoughts about what kinds of additional processing would be appropriate here for what use cases, we'd love to hear them in the comments.

Because these hourly tables are quite long, they are being distributed as Parquet files rather than in the SQLite database.

Additional documentation can be found in our [EIA-930 data source docs](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_sources/eia930.html).  Examples of how to work with this data and some graphical sanity checks can be found in [this Jupyter notebook on Kaggle](https://www.kaggle.com/code/catalystcooperative/eia-930-sanity-checks)

* [core_eia930__hourly_interchange](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-eia930-hourly-interchange)
* [core_eia930__hourly_net_generation_by_energy_source](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-eia930-hourly-net-generation-by-energy-source)
* [core_eia930__hourly_operations](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-eia930-hourly-operations)
* [core_eia930__hourly_subregion_demand](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-eia930-hourly-subregion-demand)

## EIA AEO

We've integrated a handful of tables from [EIA's Annual Energy Outlook](https://www.eia.gov/outlooks/aeo/). Additional tables are relatively easy to integrate, so if there's something you'd like to have available, let us know in the comments.

* [core_eiaaeo__yearly_projected_electric_sales](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-eiaaeo-yearly-projected-electric-sales)
* [core_eiaaeo__yearly_projected_generation_in_electric_sector_by_technology](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-eiaaeo-yearly-projected-generation-in-electric-sector-by-technology)
* [core_eiaaeo__yearly_projected_generation_in_end_use_sectors_by_fuel_type](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-eiaaeo-yearly-projected-generation-in-end-use-sectors-by-fuel-type)

## NREL ATB

We've also pulled in [NREL's Annual Technology Baseline](https://www.nrel.gov/analysis/data-tech-baseline.html). We have a few lingering questions about some of these tables, which can be found [in this comment](https://github.com/catalyst-cooperative/pudl/pull/3630#pullrequestreview-2053102112). If you find anything odd please let us know.

* [core_nrelatb__yearly_projected_cost_performance](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-nrelatb-yearly-projected-cost-performance)
* [core_nrelatb__yearly_projected_financial_cases](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-nrelatb-yearly-projected-financial-cases)
* [core_nrelatb__yearly_projected_financial_cases_by_scenario](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-nrelatb-yearly-projected-financial-cases-by-scenario)
* [core_nrelatb__yearly_technology_status](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-nrelatb-yearly-technology-status)

## GridPath RA Toolkit

We've integrated a small subset of the data that went into [the GridPath Resource Adequacy Toolkit](https://gridlab.org/gridpathratoolkit/), namely hourly available capacity factors for wind and solar generators, spatially aggregated to the balancing area. There's much more data that could be integrated here, including hourly estimates of weather-dependent generator availability for thermal generators as well as renewables.

More details can be found in our [GridPath RA Toolkit data source documentation](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_sources/gridpathratoolkit.html)

* [core_gridpathratoolkit__assn_generator_aggregation_group](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#core-gridpathratoolkit-assn-generator-aggregation-group)
* [out_gridpathratoolkit__hourly_available_capacity_factor](https://catalystcoop-pudl.readthedocs.io/en/nightly/data_dictionaries/pudl_db.html#out-gridpathratoolkit-hourly-available-capacity-factor)

**Comment by jdangerx** [comment]
The release is live!

Zenodo: https://zenodo.org/records/11292273
AWS S3: s3://pudl.catalyst.coop/v2024.5.0
Google Cloud Storage (requester pays): gs://pudl.catalyst.coop/v2024.5.0

ReadTheDocs: https://catalystcoop-pudl.readthedocs.io/en/v2024.5.0/release_notes.html
PyPI: https://pypi.org/project/catalystcoop.pudl/2024.5.0/
